File: app/src/main/java/com/metahumanz/pacilread/theme/ThemedActivity.java
package com.metahumanz.pacilread.theme;

import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;

import androidx.appcompat.app.AppCompatActivity;

import com.metahumanz.pacilread.ui.EdgeToEdgeHelper;

public abstract class ThemedActivity extends AppCompatActivity {
    private String appliedThemeBucket = ThemeModeHelper.MODE_LIGHT;
    private String appliedStyleVariant = ThemeModeHelper.LIGHT_STYLE_YUNBAI;

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(ThemeModeHelper.wrapForApp(newBase));
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(ThemeModeHelper.resolveAppThemeResId(this));
        appliedThemeBucket = ThemeModeHelper.getResolvedAppBucket(this);
        appliedStyleVariant = ThemeModeHelper.getResolvedAppStyleVariant(this);
        super.onCreate(savedInstanceState);
        EdgeToEdgeHelper.configure(this);
    }

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        EdgeToEdgeHelper.applySystemBarPaddingToContentRoot(this);
    }

    @Override
    public void setContentView(View view) {
        super.setContentView(view);
        EdgeToEdgeHelper.applySystemBarPaddingToContentRoot(this);
    }

    @Override
    public void setContentView(View view, ViewGroup.LayoutParams params) {
        super.setContentView(view, params);
        EdgeToEdgeHelper.applySystemBarPaddingToContentRoot(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        String desiredBucket = ThemeModeHelper.getResolvedAppBucket(this);
        String desiredStyleVariant = ThemeModeHelper.getResolvedAppStyleVariant(this);
        if (!desiredBucket.equals(appliedThemeBucket) || !desiredStyleVariant.equals(appliedStyleVariant)) {
            recreate();
        }
    }

    protected boolean isDarkAppTheme() {
        return ThemeModeHelper.isDark(getResources());
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/theme/ThemedActivity.kt
package com.metahumanz.pacilread.theme

import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import com.metahumanz.pacilread.ui.EdgeToEdgeHelper

abstract class ThemedActivity : AppCompatActivity() {
    private var appliedThemeBucket: String = ThemeModeHelper.MODE_LIGHT
    private var appliedStyleVariant: String = ThemeModeHelper.LIGHT_STYLE_YUNBAI

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(ThemeModeHelper.wrapForApp(newBase))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(ThemeModeHelper.resolveAppThemeResId(this))
        appliedThemeBucket = ThemeModeHelper.getResolvedAppBucket(this)
        appliedStyleVariant = ThemeModeHelper.getResolvedAppStyleVariant(this)
        super.onCreate(savedInstanceState)
        EdgeToEdgeHelper.configure(this)
    }

    override fun setContentView(layoutResID: Int) {
        super.setContentView(layoutResID)
        EdgeToEdgeHelper.applySystemBarPaddingToContentRoot(this)
    }

    override fun setContentView(view: View?) {
        super.setContentView(view)
        EdgeToEdgeHelper.applySystemBarPaddingToContentRoot(this)
    }

    override fun setContentView(view: View?, params: ViewGroup.LayoutParams?) {
        super.setContentView(view, params)
        EdgeToEdgeHelper.applySystemBarPaddingToContentRoot(this)
    }

    override fun onResume() {
        super.onResume()
        val desiredBucket = ThemeModeHelper.getResolvedAppBucket(this)
        val desiredStyleVariant = ThemeModeHelper.getResolvedAppStyleVariant(this)
        if (desiredBucket != appliedThemeBucket || desiredStyleVariant != appliedStyleVariant) recreate()
    }

    protected fun isDarkAppTheme(): Boolean = ThemeModeHelper.isDark(resources)
}
